src/librustc/session/mod.rs

```diff
@@ -371,20 +370,11 @@ impl Session {
             }
         };
 
-        match self.opts.error_format {
-            // when outputting JSON for tool consumption, the tool might want
-            // the duplicates
-            config::ErrorOutputType::Json(_) => {
-                do_method()
-            },
-            _ => {
-                let lint_id = DiagnosticMessageId::LintId(lint::LintId::of(lint));
-                let id_span_message = (lint_id, span, message.to_owned());
-                let fresh = self.one_time_diagnostics.borrow_mut().insert(id_span_message);
-                if fresh {
-                    do_method()
```

Comment:
The motivation for defining the `do_method` closure was to avoid repeating the `match method` code for the JSON and human-friendly cases: if we're going to start treating these the same, we can delete lines 362–371 and put the match here.

Reply:
fixed